synapse/lib/auth.py

```diff
@@ -1282,7 +1285,7 @@ async def setAdmin(self, admin, gateiden=None, logged=True):
         if not isinstance(admin, bool):
             raise s_exc.BadArg(mesg='setAdmin requires a boolean')
 
-        if self == self.auth.rootuser and not admin:
+        if self.iden == self.auth.rootuser.iden and not admin:
```

Comment:
```suggestion
        if self.iden == self.auth.rootuser.iden:
```
I thought root couldn't have it's admin status revoked?